### PR TITLE
Many fixes to get unit tests passing on Windows.

### PR DESCRIPTION
**Pull request by rkimball**
Not everything passes, but much more passes than used to.
* Fixed some compiler warnings
* nvcc working on Windows
* CUDA codegen tests passing

### CODE REVIEW

File: python/tvm/contrib/cc.py [hunk 1]
```diff
@@ -212,9 +214,9 @@ def _linux_compile(output, objects, options, compile_cmd="g++", compile_shared=F
         raise RuntimeError(msg)
 
 
-def _windows_shared(output, objects, options):
+def _windows_compile(output, objects, options):
     cmd = ["clang"]
-    cmd += ["-O2", "-flto=full", "-fuse-ld=lld-link"]
+    cmd += ["-O2", "-v", "-fvisibility=default", "-export-all-symbols"]
```
Comment (tqchen) [inline python/tvm/contrib/cc.py:219]: Can we check if we need `-export-all-symbols` after https://github.com/apache/tvm/pull/6698? I believe we should have marked the life scopes correctly.

**Review by tqchen — CHANGES_REQUESTED** [review]

File: python/tvm/contrib/cc.py (hunk 1 above)
Comment (rkimball) [reply python/tvm/contrib/cc.py:219]: Yes, there are some debug flags in there too, -v. That line needs to be cleaned up.

Comment (tqchen) [reply python/tvm/contrib/cc.py:219]: OK, thanks, please add a comment about the need of debug flags there.

Comment (rkimball) [reply python/tvm/contrib/cc.py]: The current code in the PR only has the flag `-O2`. The other flags are not required.

**Review by tqchen — APPROVED** [review]

**Comment by tqchen** [comment]
Thanks @rkimball !

**Pull request merged**